### PR TITLE
Add sanctum token auth support

### PR DESCRIPTION
**Pull request by allnightru**
Add support for the token-based sanctum authentication.
Implementation inspired by nuxt-auth-sanctum module.
- Added config parameter: echo.authentication.mode, "cookie" or "token".
- Added support for external tokenStorage (implementation from nuxt-auth-sanctum module)
- Changed "onRequest" hook to send Bearer token if authentication.mode is "token".

### CODE REVIEW

**Comment by manchenkoff** [comment]
Hey @allnightru, thanks for the contribution! I'll try to review this one today, in the meantime, pls check linter issues

File: src/runtime/types.ts [hunk 1]
```diff
@@ -24,6 +29,10 @@ export interface Authentication {
    * @default 'X-XSRF-TOKEN'
    */
   csrfHeader?: string
+  /**
+   * Token storage handlers to be used by the client.
+   */
+  tokenStorage?: TokenStorage
```
Comment (manchenkoff) [inline src/runtime/types.ts:35]: I think this is supposed to be used as app options interface and not ModuleOptions, you can check how it is defined in `nuxt-auth-sanctum` as well

**Comment by manchenkoff** [comment]
Closes #44

File: src/runtime/plugin.client.ts [hunk 2]
```diff
@@ -35,24 +35,34 @@ function createFetchClient(
...
-
-      if (!csrfToken.value) {
-        await $fetch(authentication.csrfEndpoint, {
-          baseURL: authentication.baseUrl,
-          credentials: 'include',
-          retry: false,
-        })
-
-        csrfToken = readCsrfCookie(authentication.csrfCookie)
+      // todo: move this to interceptors
```
Comment (manchenkoff) [inline src/runtime/plugin.client.ts:38]: interceptors are not really needed here unless we have more layers to apply, so I would suggest either removing todo comments or creating a new issue instead

**Review by manchenkoff — CHANGES_REQUESTED** [review]
minor changes and fixing linter issues are needed

**Comment by ivan-plotnikov** [comment]
+1

**Review by manchenkoff — APPROVED** [review]

**Pull request merged**